.github/actions/cluster/lxd-cluster.sh

```diff
@@ -0,0 +1,292 @@
+        done
+}
+
+# instanceIPv4 returns the IPv4 address of the instance with the given name.
+instanceIPv4() {
+        instance="$1"
+
+        # Try for enp5s0 (VM) and eth0 (container) interfaces.
+        for inf in enp5s0 eth0; do
+                ipv4=$(lxc ls "${instance}" -f csv -c 4 | grep -oP "(\d{1,3}\.){3}\d{1,3}(?= \(${inf}\))" || true)
+                if [ "${ipv4}" != "" ]; then
+                        echo "${ipv4}"
+                        return
+                fi
+        done
+
+        echo "Error: Failed to obtain IPv4 address of instance ${instance}"
+        return 1
+}
+
+
+#========================
+# Cluster setup
+#========================
+
+# deploy deploys instances required for a LXD cluster.
+deploy() {
+        # Create dedicated network.
+        echo "Creating network ${NETWORK_NAME} ..."
+        exists=$(lxc network list --format csv | awk -F, '{print $1}' | grep "${NETWORK_NAME}" || true)
+        if [ ! "${exists}" ]; then
+                lxc network create "${NETWORK_NAME}"
+        fi
+
+        # Create storage pool.
+        echo "Creating storage pool ${STORAGE_POOL} ..."
+        exists=$(lxc storage list --format csv | awk -F, '{print $1}' | grep "${STORAGE_POOL}" || true)
+        if [ ! "${exists}" ]; then
+                lxc storage create "${STORAGE_POOL}" zfs
+        fi
+
+        # Setup cluster VMs.
+        for i in $(seq 1 "${CLUSTER_SIZE}"); do
+                instance="${INSTANCE}-${i}"
+
+                echo "Creating instance ${instance} ..."
+                exists=$(lxc list --format csv --columns n "${instance}")
+                if [ "${exists}" != "" ]; then
+                        echo "Instance ${instance} is already running"
+                        continue
+                fi
+
+                args=""
+                if [ "${INSTANCE_TYPE}" = "virtual-machine" ]; then
+                        args="--vm"
+                else
+                        args="-c security.nesting=true"
+                fi
+
+                lxc launch "${INSTANCE_IMAGE}" "${instance}" \
+                        --storage "${STORAGE_POOL}" \
+                        --network "${NETWORK_NAME}" \
+                        -c limits.cpu=4 \
+                        -c limits.memory=4GiB \
+                        $args
+        done
+
+        # Wait for instances to become ready.
+        for i in $(seq 1 "${CLUSTER_SIZE}"); do
+                instance="${INSTANCE}-${i}"
+                waitInstance "${instance}"
+                lxc exec "${instance}" -- systemctl is-system-running --wait
+        done
+
+        # Install LXD on VMs.
+        for i in $(seq 1 "${CLUSTER_SIZE}"); do
+                instance="${INSTANCE}-${i}"
+
+                echo "Preparing instance ${instance} ..."
+
+                # Install snap daemon.
+                lxc exec "${instance}" --env=DEBIAN_FRONTEND=noninteractive -- bash -c "\
+                        apt-get -qq -y install snapd" > /dev/null
```

Comment:
There should be no need to wrap it with a `bash` command:
```suggestion
                lxc exec "${instance}" --env=DEBIAN_FRONTEND=noninteractive -- apt-get -qq -y install snapd > /dev/null
```